src/network/networkd-wwan.c

```diff
@@ -0,0 +1,666 @@
+                if (address->source != NETWORK_CONFIG_SOURCE_MODEM_MANAGER)
+                        continue;
+
+                if (!address_is_marked(address))
+                        continue;
+
+                r = address_remove(address, link);
+                if (r < 0)
+                        ret = r;
+        }
+
+        SET_FOREACH(route, link->manager->routes) {
+                if (route->source != NETWORK_CONFIG_SOURCE_MODEM_MANAGER)
+                        continue;
+
+                if (!route_is_marked(route))
+                        continue;
+
+                r = route_remove(route, link->manager);
+                if (ret)
+                        ret = r;
+        }
+
+        if (ret < 0)
+                return ret;
+
+        if (link->bearer_messages == 0)
+                link->bearer_configured = true;
+
+        if (!link->bearer_configured)
+                link_set_state(link, LINK_STATE_CONFIGURING);
+
+        link_check_ready(link);
+
+        return ret;
+}
+
+int link_apply_bearer(Link *link) {
+        Bearer *b = NULL;
+        int r;
+
+        assert(link);
+
+        (void) link_get_bearer(link, &b);
+
+        r = link_apply_bearer_impl(link, b);
+        if (r < 0)
+                link_enter_failed(link);
+
+        return r;
+}
+
+int bearer_update_link(Bearer *b) {
+        Link *link;
+        int r;
+
+        assert(b);
+        assert(b->modem);
+        assert(b->modem->manager);
+
+        if (!b->name)
+                return 0;
+
+        if (link_get_by_name(b->modem->manager, b->name, &link) < 0)
+                return 0;
+
+        r = link_reconfigure_impl(link, 0);
+        if (r < 0)
+                link_enter_failed(link);
+        if (r != 0) /* r > 0 means interface is reconfigured. */
+                return r;
+
+        r = link_apply_bearer_impl(link, b);
+        if (r < 0)
+                link_enter_failed(link);
+
+        /*
+         * Need to bring up the interface after the modem has connected.
+         * This is because ModemManger does the following while connecting:
+         * <msg> [1755871777.322239] [modem2] state changed (registered -> connecting)
+         * <dbg> [1755871777.325012] [modem2/bearer5] launching connection with QMI port (cdc-wdm0) and data port (wwan0) (multiplex none)
+         * <dbg> [1755871777.327665] [cdc-wdm0/qmi] bringing down data interface 'wwan0'
+         * <dbg> [1755871777.330108] [modem2/wwan0/net] interface index: 9
+         * <dbg> [1755871777.335265] [cdc-wdm0/qmi] deleting all links in data interface 'wwan0'
+         */
+
+        r = link_request_to_bring_up_or_down(link, b->connected);
+        if (r < 0)
+                link_enter_failed(link);
+
+        return r;
+}
+
+void bearer_drop(Bearer *b) {
+        assert(b);
+
+        b->connected = false;
+        b->apn = mfree(b->apn);
+
+        (void) bearer_update_link(b);
+
+        bearer_free(b);
+}
+
+void modem_drop(Modem *modem) {
```

Comment:
I do not think this is necessary. Please simply use modem_free().

Reply:
ok